src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -1755,6 +1756,151 @@ class TypeSpecTest {
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("avacado", String::class)
+            .build())
+        .addProperty(PropertySpec.builder("avacado", String::class)
+            .initializer("avacado")
+            .build())
+        .addSuperinterface(InlineSuperInterface::class)
+        .addModifiers(INLINE)
+        .build()
+
+    assertThat(guacamole.toString()).isEqualTo("""
+      |inline class Guacamole(val avacado: kotlin.String) : com.squareup.kotlinpoet.TypeSpecTest.InlineSuperInterface
+      |""".trimMargin())
+  }
+
+  @Test fun inlineClassWithTwoParameters() {
+    assertThrows<IllegalStateException> {
+      TypeSpec.classBuilder("Guacamole")
+          .primaryConstructor(FunSpec.constructorBuilder()
+              .addParameter("avacado", String::class)
+              .addParameter("garlic", String::class)
+              .build())
+          .addModifiers(INLINE)
+          .build()
+    }.hasMessageThat().isEqualTo("Inline class must have 1 parameter in constructor")
+  }
+
+  @Test fun inlineClassWithoutProperties() {
+    assertThrows<IllegalStateException> {
+      TypeSpec.classBuilder("Guacamole")
+          .primaryConstructor(FunSpec.constructorBuilder()
+              .addParameter("avacado", String::class)
+              .build())
+          .addModifiers(INLINE)
+          .build()
+    }.hasMessageThat().isEqualTo("Inline class must have a single read-only (val) property.")
+  }
+
+  @Test fun inlineClassWithMutableProperties() {
+    assertThrows<IllegalStateException> {
+      TypeSpec.classBuilder("Guacamole")
+          .primaryConstructor(FunSpec.constructorBuilder()
+              .addParameter("avacado", String::class)
+              .build())
+          .addProperty(PropertySpec.builder("avacado", String::class)
+              .initializer("avacado")
+              .mutable(true)
+              .build())
+          .addModifiers(INLINE)
+          .build()
+    }.hasMessageThat().isEqualTo("Inline class must have a single read-only (val) property.")
+  }
+
+  @Test fun inlineClassWithPrivateConstructor() {
+    assertThrows<IllegalStateException> {
+      TypeSpec.classBuilder("Guacamole")
+          .primaryConstructor(FunSpec.constructorBuilder()
+              .addParameter("avacado", String::class)
+              .addModifiers(PRIVATE)
+              .build())
+          .addProperty(PropertySpec.builder("avacado", String::class)
+              .initializer("avacado")
+              .build())
+          .addModifiers(INLINE)
+          .build()
+    }.hasMessageThat().isEqualTo("Inline class must have a public primary constructor")
+  }
+
+  @Test fun inlineEnumClass() {
+    val guacamole = TypeSpec.enumBuilder("Foo")
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("x", Int::class)
+            .build())
+        .addEnumConstant("A", TypeSpec.anonymousClassBuilder()
+            .addSuperclassConstructorParameter("%S", "1")
```

Comment:
supernit: Constructor param's type is `Int`, why are we passing a `String`?

Reply:
Woops. My bad